pkg/patterns/addon/patch.go

```diff
@@ -35,25 +35,38 @@ import (
 func ApplyPatches(ctx context.Context, object declarative.DeclarativeObject, objects *manifest.Objects) error {
 	log := log.Log
 
-	p, ok := object.(addonsv1alpha1.Patchable)
-	if !ok {
-		return fmt.Errorf("provided object (%T) does not implement Patchable type", object)
-	}
-
+	var p addonsv1alpha1.Patchable
```

Comment:
Nit: Do you need to declare this here?  Looks like you could declare it in on line 54 with `else if p, ok := object....` ?